File: php/features.php
<?php
declare(strict_types=1);

// NekosiaNeko SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NekosiaNekoFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NekosiaNekoBaseFeature();
            case "test":
                return new NekosiaNekoTestFeature();
            default:
                return new NekosiaNekoBaseFeature();
        }
    }
}

File: rb/features.rb
# NekosiaNeko SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NekosiaNekoFeatures
  def self.make_feature(name)
    case name
    when "base"
      NekosiaNekoBaseFeature.new
    when "ratelimit"
      NekosiaNekoRatelimitFeature.new
    when "retry"
      NekosiaNekoRetryFeature.new
    when "test"
      NekosiaNekoTestFeature.new
    when "timeout"
      NekosiaNekoTimeoutFeature.new
    else
      NekosiaNekoBaseFeature.new
    end
  end
end
